### ISSUE TITLE
Duration add/subtract can break humanize with mixed units

### ISSUE THREAD

**Issue by hahn-kev** [moment/moment#4641]
When creating a duration of days and then trying to subtract years, once I call humanize it looks like it added the number of years instead. However creating a duration with years and subtracting years works just fine.

Example:
```js
console.log(moment.duration(2, 'years')
            .subtract(1, 'years')
            .humanize()
           );
console.log(moment.duration(730.5, 'days')
            .subtract(1, 'years')
            .humanize()
           );
```
These should print the same things, but at the moment the first one prints `'a year'` as you would expect but the second one prints `'3 years'`

**Environment:**
Chrome 66, Windows
Moment 2.22.1

**Comment by ichernev** [comment]
This is a glaring bug: after mixing some signs in moment.duration abs could abruptly change the interval.

```javascript
moment.duration(730.5, 'days').subtract(1, 'years')
> 730.5 days forward, 1 year backward (total 1 year)
moment.duration(730.5, 'days').subtract(1, 'years').abs
> 730.5 days forward, 1 year forward (total 3 years)
```

**Comment by hahn-kev** [comment]
Looks like this same issue has come up before #2166 Unit tests should probably be made for this, it's not even a corner case.

**Comment by ichernev** [comment]
@hahn-kev yes, it is related. It just needs to happen with abs too. Or maybe abs can just switch the signs if necessary -- that will preserve the +/- situation. The duration code needs to be examined if the rest of the methods are OK with that :)

The gist is that bubbling destroys that information, but keeping it might make sense. "Add 2 days - 2 minutes" will be executed exactly as this, instead of "add N milliseconds", which behaves differently across DST.
